internal/http/services/owncloud/ocs/handlers/cloud/capabilities/uploads.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package capabilities
+
+import (
+	"strings"
+
+	"github.com/cs3org/reva/internal/http/services/owncloud/ocs/data"
+)
+
+type chunkProtocol string
+
+var (
+	chunkV1  chunkProtocol = "v1"
+	chunkNG  chunkProtocol = "ng"
+	chunkTUS chunkProtocol = "tus"
+)
+
+func (h *Handler) getCapabilitiesForUserAgent(userAgent string) data.CapabilitiesData {
+	if userAgent != "" {
+		for k, v := range h.userAgentChunkingMap {
+			// we could also use a regexp for pattern matching
+			if strings.Contains(userAgent, k) {
+				// Creating a copy of the capabilities struct is less expensive than taking a lock
+				c := h.c
+				setCapabilitiesForChunkProtocol(chunkProtocol(v), &c)
+				return c
+			}
+		}
+	}
+	return h.c
+}
+
+func setCapabilitiesForChunkProtocol(cp chunkProtocol, c *data.CapabilitiesData) {
+	switch cp {
+	case chunkV1:
+		// 2.7+ will use Chunking V1 if "capabilities > files > bigfilechunking" is "true" AND "capabilities > dav > chunking" is not there
+		c.Capabilities.Files.BigFileChunking = true
+		c.Capabilities.Dav.Chunking = ""
```

Comment:
In this case the json output cannot contain c.Capabilities.Dav.Chunking, in this case it will output an "empty" string but the key will still be there.

Doing:
```
c.Capabilities.Dav = nil
```

should work

Reply:
That would also remove some of the required DAV capabilities https://github.com/cs3org/reva/blob/master/internal/http/services/owncloud/ocs/handlers/cloud/capabilities/capabilities.go#L122-L127

I tested v2.7.0 with this code and it did use v1 chunking with this config, so I think that should be good.

Reply:
We don't use the other dav parameters, so it should be also safe to remove it completly.

Reply:
Will do